cognite_toolkit/_cdf_tk/utils/http_client/_client.py

```diff
@@ -152,7 +152,7 @@ def _create_headers(self) -> MutableMapping[str, str]:
         headers["accept"] = "application/json"
         headers["x-cdp-sdk"] = f"CogniteToolkit:{get_current_toolkit_version()}"
         headers["x-cdp-app"] = self._config.client_name
-        headers["cdf-version"] = self._config.api_subversion
+        headers["cdf-version"] = api_version or self._config.api_subversion
```

Comment:
As requests ignore any header values that are None, this shouldn't be an actual problem.